extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -563,11 +563,14 @@ static JsonScanFormat autoDetect(main::ClientContext* context, const std::string
         auto next = std::min<uint64_t>(numTuplesRead, numRowsToDetect);
         yyjson_val *key = nullptr, *ele = nullptr;
         for (auto i = 0u; i < next; i++) {
-            const auto& val = localState.values[i];
-            auto objIter = yyjson_obj_iter_with(val);
+            auto* doc = localState.docs[i];
+            if (nullptr == doc) {
+                continue;
+            }
```

Comment:
can this be an assertion or is the null check required?

Reply:
I think is should be an assertion, doc should never be null